src/listener.coffee

```diff
@@ -39,7 +39,7 @@ class TextListener extends Listener
   # callback - A Function that is triggered if the incoming message matches.
   constructor: (@robot, @regex, @callback) ->
     @matcher = (message) =>
-      if message instanceof TextMessage
+      if message.match
```

Comment:
I think my brain was broken yesterday. I no longer have strong feelings about this. Checking for a specific type makes the minimal behavior change though, so the current state is probably the right one.